### DISCUSSION TITLE
Generate thumbnails using smart crop sometimes fail miserably when using eg animals

### DISCUSSION THREAD

**Discussion by marcuslindblom** [imgproxy/imgproxy#1146] [Q&A]
We have this media library and I want to create good looking thumbnails for most of the images that are uploaded. I have activated advanced smart crop in my container and this is the parameters I'm using `/w:240/h:135/dpr:2/rt:fill/q:35/gravity:sm/`

The result with [this source](https://www.pexels.com/photo/photo-of-fox-sitting-on-ground-2295744/)  looks like this.

<img width="374" alt="image" src="https://user-images.githubusercontent.com/319720/229276725-7dd964f1-e37b-4009-99a7-57fda19b2ef3.png">

This view is from our library and it does not do a good job. The first image has a car in the picture but that is not visible at all. Is it possible to fix this using some other parameters or do you have any other suggestions?

<img width="822" alt="image" src="https://user-images.githubusercontent.com/319720/229276772-69e5cbba-da31-4d5e-bf73-585b5c7a2193.png">

**Comment by DarthSim** [comment]
Both smart crop and advanced smart crop are not content-aware. They both work on the pixel level using different techniques. This means that they don't see any difference between a car, a tree, a fox, or a wall. The only thing that matters is if the pixels meet the requirements. For example, here's the features map of the fox photo:

![corners](https://user-images.githubusercontent.com/380856/229289897-32fb7631-2f49-4997-96f9-d78084002445.png)

As you can see, the asphalt under the fox's paws is noisy, so it catches the detector's attention. Such algorithms are doing a good job when you're dealing with images containing objects on a quite solid background or photos with good bokeh. But when the entropy of a whole image is uniform, such algorithms will most probably fail.

If you want a content-aware crop, you should probably take a look at the [object detection](https://docs.imgproxy.net/object_detection) feature. Though the default model that is delivered with imgproxy Pro is trained only on faces, you can use any pre-trained DarkNet YOLOv4 model with imgproxy or you can train your own. Drop us an email if you need any help with object detection.
